compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1443,7 +1435,26 @@ impl Target {
     }
 
     pub fn is_abi_supported(&self, abi: Abi) -> bool {
-        abi.generic() || !self.unsupported_abis.contains(&abi)
+        use Abi::*;
+        match abi {
+            Rust
+            | C { .. }
+            | System { .. }
+            | RustIntrinsic
+            | RustCall
+            | PlatformIntrinsic
+            | Unadjusted
+            | EfiApi => true,
+            Cdecl | Stdcall { .. } | Fastcall | Thiscall { .. } => self.arch == "x86",
+            Vectorcall | X86Interrupt => ["x86", "x86_64"].contains(&&self.arch[..]),
+            Aapcs | CCmseNonSecureCall => ["arm", "aarch64"].contains(&&self.arch[..]),
+            Win64 | SysV64 => self.arch == "x86_64",
+            PtxKernel => self.arch == "nvptx64",
+            Msp430Interrupt => self.arch == "msp430",
+            AmdGpuKernel => false, // FIXME: we don't have any built-in targets utilising this ABI
```

Comment:
```suggestion
            AmdGpuKernel => self.arch == "amdgcn",
```
Although given the state of https://github.com/rust-lang/rust/issues/51575 it looks like custom AMD GPU targets cannot currently exist either.